### DISCUSSION TITLE
Customizing Email Templates

### DISCUSSION THREAD

**Discussion by theotillotson** [impress-org/givewp#6643] [Developer Q&A] [answered]
When customizing the email templates in /give/templates/emails/ since these are core plugin files, will these be overwritten when the plugin updates?

Feature Suggestion: I would suggest adding filter hooks in header-default.php to each of the style variables similar to what is used in footer-default.php. This could be optimized into a single template file (string) that could be filtered prior to processing by Give_Emails::build_email() which would give developers full control over the email templates without needing to modify any core template files directly.

Changing the width of the branding image in the email was a hassle but could be done with a filter.

```
if ( class_exists('Give_Email_Notification_Util')) {
	add_filter('give_email_message', 'dcpci_filter_give_email_message_logo', 11, 2);

	function dcpci_filter_give_email_message_logo($message, $obj) {
		$needle = '<div id="template_header_image"><p style="margin-top:0;"><img style="max-width:450px;" ';
		$replace = '<div id="template_header_image"><p style="margin:0px; margin-block-end: -3px;"><img style="width:100%;height:auto;max-width:642px;" ';
		return str_replace( $needle, $replace, $message );
	}
}
```

**Comment by kjohnson** [comment, marked as answer]
@theotillotson GiveWP emails use the WordPress templating system, which allows for customizing templates without the changes being lost during a plugin update.

Here are some details on how you can customize the email templates in GiveWP: https://givewp.com/documentation/core/settings/emails/customizing-email-templates/
